packages/bitcore-wallet-service/src/lib/pushnotificationsservice.ts

```diff
@@ -287,15 +290,38 @@ export class PushNotificationsService {
   _getRecipientsList(notification, notificationType, cb) {
     if (notificationType.broadcastToActiveUsers) return cb(null, []);
 
-    this.storage.fetchWallet(notification.walletId, (err, wallet) => {
+    this.storage.fetchWallet(notification.walletId, async (err, wallet) => {
       if (err) return cb(err);
       if (!wallet) return cb(null, []);
 
       let unit;
+      let tokenId;
+      let tokenName;
+      let tokenDecimals;
       if (wallet.coin != Defaults.COIN) {
         unit = wallet.coin;
       }
 
+      if (wallet.isSlpToken) {
+        let url = _.get(config.supportToken[unit], 'chronikClientUrl', undefined);
+        if (!url) return cb(`chronik not support ${unit}`);
+        const chronikClient = new ChronikClient(url);
```

Comment:
I'm not sure but maybe we can create a new chronik client at the class level instead of creating new client for each notification? If there're thousand of notifications then we will create thousands of clients? Is it a good solution?